### ISSUE TITLE
Logger

### ISSUE THREAD

**Issue by Volvoxpl** [szumel/rusher#1]
I suggest adding a logger to log all actions done by steps instead of printing them to terminal.
There will be more control over the output (send mail when error, display more/less information, etc.)

**Comment by szumel** [comment]
There are two cases. First one is handling errors, second one is handling the steps information during execution.

I think we should use standard library Logger for handling errors. Wrap it with some func like Log(error).
In second case my proposition is to create some buffer to which steps will write information about step's state.

In the end of steps execution we could call to write/send buffer somewhere.
We could create State struct which implements io.Writer, create method sth like State.Inform()

Something like:
`
type State struct {
    buffer []byte
}
`
I am going to create some brief, so we will discuss this after that.

**Comment by szumel** [comment]
@Volvoxpl any ideas? thoughts?

**Comment by Volvoxpl** [comment]
Yeah, 2 outputs (error and standard) seem to be OK. I'm still mentally in PHP where you need to output as soon as possible in case of fatal error ;)
